src/libs/Navigation/AppNavigator/BaseDrawerNavigator.js

```diff
@@ -34,46 +34,63 @@ const propTypes = {
 };
 const Drawer = createDrawerNavigator();
 
-const BaseDrawerNavigator = (props) => {
-    const content = (
-        <Drawer.Navigator
-            backBehavior="none"
-            key={`BaseDrawerNavigator${props.isSmallScreenWidth}`}
-            defaultStatus={Navigation.getDefaultDrawerState(props.isSmallScreenWidth)}
-            sceneContainerStyle={styles.navigationSceneContainer}
-            drawerContent={props.drawerContent}
-            screenOptions={{
-                cardStyle: styles.navigationScreenCardStyle,
-                headerShown: false,
-                drawerType: StyleUtils.getNavigationDrawerType(props.isSmallScreenWidth),
-                drawerStyle: StyleUtils.getNavigationDrawerStyle(
-                    props.isSmallScreenWidth,
-                ),
-                swipeEdgeWidth: 500,
-            }}
-        >
-            {_.map(props.screens, screen => (
-                <Drawer.Screen
-                    key={screen.name}
-                    name={screen.name}
-                    component={screen.component}
-                    initialParams={screen.initialParams}
-                />
-            ))}
-        </Drawer.Navigator>
-    );
+const defaultProps = {
+    isMainScreen: false,
+};
 
-    if (!props.isMainScreen && !props.isSmallScreenWidth) {
-        return (
-            <View style={styles.navigationSceneFullScreenWrapper}>
-                {content}
-            </View>
-        );
+class BaseDrawerNavigator extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            // calculate the defaultStatus only once on mount to prevent breaking the navigation internal state.
+            // Directly passing the dynamically calculated defaultStatus to drawer Navigator breaks the internal state
+            // And prevents the Drawer actions from reaching to action Drawer Navigation while screen is resized on from Web to mobile Web.
+            defaultStatus: Navigation.getDefaultDrawerState(props.isSmallScreenWidth),
```

Comment:
Small comments here:

// Calculate

Can you link to an issue regarding comment #2 if it exists?

// Prevents drawer action

Reply:
I will update it after your testing. But I don't have any links to the issue that I have mentioned here. Here is the explanation Why.

This component uses a key prop to remount the navigator when the browser resizes. This hack was done to prevent the layout from breaking on the drawer. Now due to that, it is somehow breaking the internal navigation state on change of defaultStatus.  Rest of the explanation is in linked proposal.